### PR TITLE
Remove unused repository in composer.json

### PR DESCRIPTION
**Pull request by Art4**
The youthweb/urllinker library is now fully typed in the latest version. Since adding return types to interface methods is a breaking change, the major version has been raised. In addition, deprecated methods have been removed, but they are not used in nextcloud/mails. Therefore, further changes are not necessary.

<details> 
<summary>Full changelog:</summary>

from https://github.com/youthweb/urllinker/blob/2.0.0/CHANGELOG.md

> ## [2.0.0](https://github.com/youthweb/urllinker/compare/1.5.1...2.0.0) - 2022-12-14
> 
> ### Changed
> 
> - UrlLinker is now fully typed.
> - **BREAKING** Return type `string` was added to method `Youthweb\UrlLinker\UrlLinkerInterface::linkUrlsAndEscapeHtml()`.
> - **BREAKING** Return type `string` was added to method `Youthweb\UrlLinker\UrlLinkerInterface::linkUrlsInTrustedHtml()`.
> 
> ### Removed
> 
> - **BREAKING** The deprecated support for `callable` in the config option `htmlLinkCreator` was removed, provide `Closure` instead
> - **BREAKING** The deprecated support for `callable` in the config option `emailLinkCreator` was removed, provide `Closure` instead
> - **BREAKING** The deprecated support for non-boolean in the config option `allowFtpAddresses` was removed, provide `boolean` instead
> - **BREAKING** The deprecated support for non-boolean in the config option `allowUpperCaseUrlSchemes` was removed, provide `boolean` instead
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::getAllowFtpAddresses()` was removed
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::setAllowFtpAddresses()` was removed
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::getAllowUpperCaseUrlSchemes()` was removed
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::setAllowUpperCaseUrlSchemes()` was removed
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::getHtmlLinkCreator()` was removed
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::setHtmlLinkCreator()` was removed
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::getEmailLinkCreator()` was removed
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::setEmailLinkCreator()` was removed
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::getValidTlds()` was removed
> - **BREAKING** The deprecated method `Youthweb\UrlLinker\UrlLinker::setValidTlds()` was removed

</details>

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
Thanks! :)

**Comment by Art4** [comment]
Looks like I have to update the `composer.lock` file too.

File: composer.lock [hunk 1]
```diff
@@ -3345,7 +3345,8 @@
     "prefer-stable": false,
     "prefer-lowest": false,
     "platform": {
-        "php": ">=7.4 <=8.1"
+        "php": ">=7.4 <=8.1",
+        "ext-openssl": "*"
```
Comment (Art4) [inline composer.lock:3349]: Not sure why this was added automatically by composer... 🤔

Comment (ChristophWurst) [reply composer.lock:3349]: @st3iny didn't update the lock file with https://github.com/nextcloud/mail/pull/7741#pullrequestreview-1213095581 :)

**Comment by Art4** [comment]
I updated the `composer.lock` file and also removed the unused repository in `composer.json`. The dependency to this repo was removed in #4836.

And I think #2028 can be closed.

**Review by kesselb — APPROVED** [review]
Thank you :+1:

**Pull request merged**